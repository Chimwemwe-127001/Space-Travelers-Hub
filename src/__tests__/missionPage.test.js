import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import MissionsPage from '../component/pages/missionpage/missionPage';

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('testing the app', () => {
  it('testing MissionsPage UI', () => {
    act(() => {
      render(
        <Provider store={store}>
          <MissionsPage />
        </Provider>, container,
      );
    });

    expect(pretty(container.innerHTML)).toMatchSnapshot();
  });
});
