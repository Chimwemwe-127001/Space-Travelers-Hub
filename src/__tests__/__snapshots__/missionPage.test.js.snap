// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the app testing MissionsPage UI 1`] = `
"<div class=\\"mission__container\\">
  <table class=\\"table\\">
    <thead>
      <tr class=\\"row\\">
        <th class=\\"column head\\">Mission</th>
        <th class=\\"column head\\">Description</th>
        <th class=\\"column head\\">Status</th>
        <th class=\\"column head\\"> </th>
      </tr>
    </thead>
    <tbody></tbody>
  </table>
</div>"
`;
